Improve code highlighting in web UI

pat-s opened the pull request:
@woodpecker-ci/maintainers as this might be another controversial design decision, please vote on the initial post of this PR with 👍 and 👎

---

This introduces a custom color. However, I found no color of the current palette to be matching here and/or creating a good contrast.
In addition, I slightly tweaked the padding and rounding values.

## Before

<img width="404" alt="image" src="https://github.com/user-attachments/assets/e825ea0d-a2e6-46e2-9384-8fd9b1a57fad">

<img width="370" alt="image" src="https://github.com/user-attachments/assets/cb338fe6-58c8-4084-bfad-71ec609b88a3" />



## After

<img width="382" alt="image" src="https://github.com/user-attachments/assets/f2025d2d-1b29-4ca7-b3b5-87a3429e5556" />

<img width="386" alt="image" src="https://github.com/user-attachments/assets/74f75d03-ac0c-4ce8-9779-70478da9e420" />

<img width="367" alt="image" src="https://github.com/user-attachments/assets/57d66200-7977-4ed2-b4ee-9feefddbed3b" />

xoxys commented:
Im not against changing the code block color but not this way. If we want a new one we should create a new color theme with `--wp-code-{100,200,300}`.

xoxys commented:
Keep in mind its used in various other places as well.

pat-s commented:
I don't think we need a full palette for a single light/dark code highlighting color.

I agree the naming is not optimal and doesn't fit into the current schema. Open for suggestions.

xoxys commented:
We dont need a full pallete right now we only have 100 and 200 just change the color of these.

xoxys commented:
We are getting lost in discussions again. At the end, I don't really care, and `wp-code` works for me as well.

pat-s commented:
Hm, I think this was just a normal discussion? You brought up some reasons, I replied and shared my thoughts. It is not even about subjective UI but code syntax, so I'd say such discussions are normal (and needed and good).

xoxys commented:
Sorry this was not meant to be offensive or rude :) The color is only used in one location. While I have some personal preferences, I don't care much enough to discuss further how it is done in detail. `wp-code` works for me we can still change it later on if there is any need for a color palette. Sorry if my last message was misleading.

pat-s commented:
Consolidated all mixed wp-code definitions into `wp-code-bg` now, which aligns with the logic of `wp-code-text`.

@xoxys Is this acceptable in your view? Yes, the color is custom and not derived from the main colour palettes but I don't think that any value from these matches well here.

Need to run the PR image first to see in which places the changes trigger a change.

xoxys commented:
Fine for me.

pat-s commented:
OK, my recent changes also change the bg color of the main log container. More work needed.

xoxys commented:
Ah yeah, I remember. That's why it's a bit questionable if we don't to keep it equal, but I'm still fine if we want to change it only for real code blocks.

pat-s commented:
@xoxys 

I refactored to keep the old color definitions and added `var(--wp-code-300)` with a custom color definitions for both light and dark. These are needed as none of the existing palette colors do even closely match a good UI outcome for these inline code boxes as they need to distinguish in a clean way from the surrounding elements.
Is the new approach OK with you?

I also updated the screenshots.

xoxys commented:
Can we not use the same color as we use for the counter badge already?

![image](https://github.com/user-attachments/assets/cdc4c8d9-5f34-4873-aa0c-239882d4972d)


This would be `colors.gray.200` and to be honest I dont see a big difference and no reason to introduce another custom color. What do you think?

pat-s commented:
@xoxys I did quite some refactoring. (Sorry for the messy diff due to the sorting.)
To summarize: I aligned the color palettes in their shades both in terms of absolute values as well as for light and dark having matching shades.
E.g. before light and dark did not have matching shades (e.g. light was only up to 300 while dark was up to 400) and the individual shades were also not matching the referenced color palette shades, making it even more confusing.

Now everything is aligned and it should be much less confusing when checking `style.css`.

You primarily want to check the changes in `style.css` and `windi` to spot these changes. For all the color related changes in the UI, deploying the PR image and inspecting visually might be the solution (I guess you already do so anyhow).

Thanks for all the work 💙️

xoxys reviewed (changes requested):
Sorry to say, but I don't really like the current state. It lacks a lot of contrast in many areas. On example:

![image](https://github.com/user-attachments/assets/1398d6b7-2ec5-4aa8-8c4d-c5953dbe9ad8)

For me also, the separator on the top is really missing now:

![image](https://github.com/user-attachments/assets/d54278a7-af71-4c3b-b5d2-cb9318cea677)

Maybe it was not a good idea to change the overall look and feel in a PR that was nearly ready to merge with the improved code blocks? :see_no_evil:

pat-s commented:
Weird I didn't catch the log background. Too many places to check I guess.

I'd say having a somewhat consistent color mapping between tailwind and the palette is worth it. 
I don't mind iterating a bit more. The state before was somewhat "wild west", I think you agree.

xoxys commented:
As I have created this wild west, I see it a bit different :D We need a very granular color pallette of gray tones to get the different layers and overlays working in terms of contrast including over effects. That's why I have picked exactly these colors. While it might look a bit random IMO it was a valid palette where 100 was the lightest value that gets darker with each value.

If we find a better pallette fine for me but I had a hard time to get contrasts working without getting too dark in the light theme and vice versa.

pat-s commented:
I'd didn't apply color changes to the palette, I mainly aligned the tailwind - palette mappings. 

This caused some tailwind classes to be different. I already updated some to recreate the old contrasts but haven't caught all yet. You're welcome to help to update the ones I missed but I can also do it myself. 

We should be able recreate the old contrasts easily. 
I also used your suggestion to use the existing gray color for the code box backgrounds.

pat-s commented:
@xoxys You can take another look. I think I was able to keep all previous colors while achieving a much more consistent palette - tailwind mapping that should ease future UI changes.

xoxys commented:
Uhm the overall look and feel is still completely different and I dont like it:

![image](https://github.com/user-attachments/assets/17913306-81dc-4d01-9d29-e145ffd67591)

![image](https://github.com/user-attachments/assets/b296eb81-8ebd-421f-b4d6-b83067cf729e)

xoxys commented:
![image](https://github.com/user-attachments/assets/bc39f317-801c-47d3-abb4-a31a8f4177bb)

On this view, the hover/non-hover color is inverted compared to the one in the screenshot above. Non-hover is white hover, is gray. This is IMO the correct behavior but should be used everywhere. The borders are way too dark now.

pat-s commented:
> Uhm the overall look and feel is still completely different and I dont like it:

I am actually confused why it looks like this now. The last time I checked it looked completely different. 
Not sure, why maybe I did too many PRs in parallel.
Just to state explicitly, that this is not the look I indented.

xoxys commented on web/src/components/layout/Settings.vue:
```diff
@@ -1,15 +1,15 @@
-    <div class="flex flex-col border-b mb-4 pb-4 justify-center dark:border-wp-background-100">
-      <div class="flex items-center justify-between">
-        <h1 class="text-xl text-wp-text-100 flex items-center gap-1">
+    <div class="flex flex-col justify-center dark:border-wp-background-200 mb-4 pb-4 border-b">
```
```suggestion
    <div class="flex flex-col justify-center dark:border-wp-background-100 mb-4 pb-4 border-b">
```

xoxys commented on web/src/components/admin/settings/queue/AdminQueueStats.vue:
```diff
@@ -1,22 +1,22 @@
   <div v-if="stats" class="flex justify-center">
     <div
-      class="bg-wp-background-200 border border-wp-background-300 dark:bg-wp-background-100 text-wp-text-100 rounded-md py-5 px-5 w-full"
+      class="border-wp-background-400 bg-wp-background-200 dark:bg-wp-background-100 px-5 py-5 border rounded-md w-full text-wp-text-100"
```
```suggestion
      class="border-wp-background-400 bg-wp-background-200 dark:bg-wp-background-300 px-5 py-5 border rounded-md w-full text-wp-text-100"
```

pat-s replied:
Verified.

xoxys reviewed (commented):
`border-wp-background-400` should be changed to `border-wp-background-500` to match the old color.

xoxys commented on web/src/style.css:
```diff
@@ -49,10 +59,12 @@
-  --wp-background-200: theme('colors.int-wp-secondary.400');
-  --wp-background-300: theme('colors.int-wp-secondary.500');
-  --wp-background-400: theme('colors.int-wp-secondary.600');
+  --wp-background-100: theme('colors.black');
```
This seems to be wrong. 100 should not be the darkest variant.

xoxys commented on web/src/components/repo/pipeline/PipelineStepList.vue:
```diff
@@ -102,14 +102,14 @@
+              class="flex items-center gap-2 border-2 hover:bg-wp-background-300 dark:hover:bg-wp-background-600 p-2 border-transparent rounded-md w-full hover-effect"
               :class="{
-                'bg-wp-background-300 dark:bg-wp-background-400': selectedStepId && selectedStepId === step.pid,
+                'bg-wp-background-400 dark:bg-wp-background-500': selectedStepId && selectedStepId === step.pid,
```
```suggestion
                'bg-wp-background-400 dark:bg-wp-background-600': selectedStepId && selectedStepId === step.pid,
```

xoxys commented:
![image](https://github.com/user-attachments/assets/224846f6-2215-46e9-b559-1ea48bae8213)

Code bocks in cli view are not covered.

pat-s commented:
> Code bocks in cli view are not covered.

This was actually on purpose. The change was only intended for inline code, not code blocks. But alright, lets apply it consistently to all code related boxes then.

xoxys commented:
Why would we use different styles for inline code blocks and multiline code blocks?

pat-s commented:
> Why would we use different styles for inline code blocks and multiline code blocks?

My main concern was the hard contrast on the inline code parts. The larger code blocks were actually "OK".

After changing the CLI code blocks, the pipeline log is now also again affected and needs to be decoupled. Unless the light one should also become light in general? (with matching line number colors etc. of course). It wouldn't look too bad with a matching lighter grey tone.

xoxys commented:
Yeah thats why I said its questionable to change it at all... however changing inlines only is weird too... However the pipeline log should stay as it if we want to change code blocks and Im ok to live with this tradeoff.

pat-s commented:
> Yeah thats why I said its questionable to change it at all.

It means work but I'd say that these classes should have been decoupled right from the start so they can more easily be changed.

> If you want to propose a light variant of the pipeline log view, feel free to do so as long as we can get enough contrast for all parts.

Maybe in a different PR.

pat-s commented:
> @pat-s looks like your last commit is not related to this PR?

Actually I just merged `main` locally. Weird it is no merge commit. 

I'll fix it.

xoxys commented on web/src/components/agent/AgentList.vue:
```diff
@@ -3,11 +3,11 @@
       v-for="agent in props.agents"
       :key="agent.id"
-      class="items-center !bg-wp-background-200 !dark:bg-wp-background-100"
+      class="items-center !bg-wp-background-200 !dark:bg-wp-background-300"
```
This change seems to be unrelated to this PR?

xoxys commented on web/src/components/atomic/ListItem.vue:
```diff
@@ -2,9 +2,9 @@
+    class="flex border-wp-background-400 bg-wp-background-100 dark:bg-wp-background-400 p-4 border rounded-md w-full overflow-hidden"
     :class="{
-      'cursor-pointer hover:shadow-md hover:bg-wp-background-300 dark:hover:bg-wp-background-300': clickable || to,
+      'cursor-pointer hover:shadow-md hover:bg-wp-background-300 dark:hover:bg-wp-background-500': clickable || to,
```
Same here.

xoxys commented on web/src/components/atomic/ListItem.vue:
```diff
@@ -2,9 +2,9 @@
     :is="to ? 'router-link' : clickable ? 'button' : 'div'"
     :to="to"
-    class="w-full flex border rounded-md bg-wp-background-100 overflow-hidden p-4 border-wp-background-400 dark:bg-wp-background-200"
+    class="flex border-wp-background-400 bg-wp-background-100 dark:bg-wp-background-400 p-4 border rounded-md w-full overflow-hidden"
```
Same here for ` dark:bg-wp-background-400`

xoxys commented on web/src/components/form/TextField.vue:
```diff
@@ -2,7 +2,7 @@
     v-if="lines === 1"
     v-model="innerValue"
-    class="w-full border border-wp-control-neutral-200 py-1 px-2 rounded-md bg-wp-background-100 focus-visible:outline-none focus-visible:border-wp-control-neutral-300"
+    class="border-wp-control-neutral-200 focus-visible:border-wp-control-neutral-300 bg-wp-background-200 px-2 py-1 border rounded-md w-full focus-visible:outline-none"
```
Same here for border and bg color

xoxys commented on web/src/components/form/TextField.vue:
```diff
@@ -11,7 +11,7 @@
     v-else
     v-model="innerValue"
-    class="w-full border border-wp-control-neutral-200 py-1 px-2 rounded-md bg-wp-background-100 focus-visible:outline-none focus-visible:border-wp-control-neutral-300"
+    class="border-wp-control-neutral-200 focus-visible:border-wp-control-neutral-300 bg-wp-background-200 px-2 py-1 border rounded-md w-full focus-visible:outline-none"
```
Same

xoxys commented on web/src/components/layout/Panel.vue:
```diff
@@ -1,18 +1,18 @@
 <template>
   <div
-    class="rounded-md w-full shadow overflow-hidden bg-wp-background-100 dark:bg-wp-background-200 border border-wp-background-400"
+    class="border-wp-background-400 bg-wp-background-100 dark:bg-wp-background-400 shadow border rounded-md w-full overflow-hidden"
```
Ill stop here :smile: please revert all border and background color changes.

pat-s commented:
@xoxys I've adjusted the color scheme ordering, which is why most bg classes are changed. 

But yeah, as I don't need these colors for the code-box adjustments, it might be easier to revert them again to keep the changes smaller.

xoxys commented on web/src/style.css:
```diff
@@ -39,8 +39,18 @@
+  --wp-code-inline-200: theme('colors.gray.200');
+
+  --wp-code-inline-text-100: theme('colors.gray.600');
+
+  /* --wp-code-300: theme('colors.int-wp-secondary.300');
+  --wp-code-600: theme('colors.int-wp-secondary.600');  */
```
```suggestion
```

pat-s commented:
Hm, something is still off with the prettier tailwind sorting in `main` and here even after merging main.

pat-s commented:
> Hm, something is still off with the prettier tailwind sorting in main and here even after merging main.

prettier tailwind sorting seems to be quite forgiving, e.g. it doesn't enforce within group-sorting and some other instances with overrides were also not accounted for in terms of being at the beginning or end of the class definition.

pat-s commented on web/src/components/form/InputField.vue:
```diff
@@ -6,7 +6,7 @@
     </div>
     <slot :id="id" />
-    <div v-if="$slots.description" class="text-wp-text-alt-100 ml-1">
+    <div v-if="$slots.description" class="text-wp-text-alt-100 ml-1 mt-1">
```
This ensures enough margin between the text and the form.

<img width="232" alt="image" src="https://github.com/user-attachments/assets/bc9ecffd-e0de-4b68-9c85-b76b33c787db" />

pat-s commented:
@xoxys FYI you're still on block here.

xoxys commented:
Yes would like to give it a final test but will unblock.
